src/engine/Server.cpp

```diff
@@ -338,8 +337,20 @@ Awaitable<void> Server::process(
       throw std::runtime_error(
           "Parameter \"query\" must not have an empty value");
     }
+    std::chrono::seconds timeLimit =
+        RuntimeParameters().get<"default-query-timeout">();
+    if (auto timeoutParam = checkParameter("timeout", std::nullopt)) {
+      std::string userTimeoutString{timeoutParam.value()};
+      // std::stoi does not support std::string_view
+      std::chrono::seconds userTimeout{std::stoi(userTimeoutString)};
+      if (userTimeout > timeLimit) {
+        // Throw exception if token is invalid
+        checkParameter("timeout", std::nullopt, accessTokenOk);
```

Comment:
1. I think we need a handwritten exception message here, because from what we currently get, the user has no change to correct their mistake.

2. This whole function can again be factored out to a named function that only returns the time limit and does all the default setting/ exception handling/parsing.